src/test/java/org/folio/circulation/resources/renewal/RegularRenewalStrategyTest.java

```diff
@@ -0,0 +1,224 @@
+package org.folio.circulation.resources.renewal;
+
+import static api.support.matchers.ResultMatchers.hasValidationError;
+import static api.support.matchers.ResultMatchers.hasValidationErrors;
+import static api.support.matchers.ValidationErrorMatchers.hasMessage;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static org.folio.circulation.domain.ItemStatus.AGED_TO_LOST;
+import static org.folio.circulation.domain.policy.Period.days;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.joda.time.DateTime.now;
+import static org.joda.time.DateTimeZone.UTC;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.folio.circulation.domain.ItemStatus;
+import org.folio.circulation.domain.Loan;
+import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.RequestQueue;
+import org.folio.circulation.domain.policy.LoanPolicy;
+import org.folio.circulation.support.results.Result;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import api.support.builders.LoanBuilder;
+import api.support.builders.LoanPolicyBuilder;
+import api.support.builders.RequestBuilder;
+import io.vertx.core.json.JsonObject;
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+@RunWith(JUnitParamsRunner.class)
+public class RegularRenewalStrategyTest {
+  @Test
+  public void canRenewLoan() {
+    final var rollingPeriod = days(10);
+    final var currentDueDate = now(UTC);
+    final var expectedDueDate = currentDueDate.plus(rollingPeriod.timePeriod());
+
+    final var loanPolicy = new LoanPolicyBuilder().rolling(rollingPeriod)
+      .renewFromCurrentDueDate().asDomainObject();
+    final var loan = new LoanBuilder().asDomainObject().changeDueDate(currentDueDate)
+      .withLoanPolicy(loanPolicy);
+
+    final var renewResult = renew(loan, null);
+
+    assertThat(renewResult.succeeded(), is(true));
+    assertThat(renewResult.value().getDueDate().getMillis(), is(expectedDueDate.getMillis()));
+  }
+
+  @Test
+  public void canCollectErrorsWhenRecallRequestedAndPolicyNorLoanableAndItemLost() {
```

Comment:
I agree that the collection aspect is important for this particular change. Is it memorable for folks coming to this code afterwards?

```suggestion
  public void cannotRenewWhenRecallRequestedAndPolicyNorLoanableAndItemLost() {
```